### ISSUE TITLE
Allow uncorrelation without derivative of energy components

### ISSUE THREAD

**Issue by limn1** [MobleyLab/alchemical-analysis#29]
There is currently no code that allows you to use the uncorrelate function with potential energy differences.

**Comment by FEPanalysis** [comment]
try "python alchemical-analysis.py -i 0"

**Comment by limn1** [comment]
Doesn't this just completely skip the step to make your correlated data uncorrelated, which is desirable if you wish to use MBAR?

**Comment by mrshirts** [comment]
I believe limn1 is correct. What the users wants to do is to use energy differences INSTEAD of dhdl's to uncorrelate the data.  

I assume you don't have dh/dl information.  There should not be that much difference between correlation times determined with energy difference data and dhdl data.

**Comment by mrshirts** [comment]
This shouldn't be that hard; one could in principle use either dhdl, du_i+1/du_i-1, or u itself to decorrelate, and leave that as an option.

**Comment by jchodera** [comment]
I would highly recommend using the reduced potential u by default (or the
equivalent superensemble \sum_k u_{s_k}(x_k) for replica exchange) unless
you have good reason to use other choices in special cases.

**Comment by mrshirts** [comment]
Well, there is a strong case, when decorrelating measurements, to use the observable that is averaged, which for TI is du/dl.  For reweighting, it's u_{i+1} - u_i (unless temperature reweighting). Using u in a system where the not-alchemically-changing energy large can result in spuriously low values, as the time correlation for the alchemical part can be slower than the energy sloshing, and get swamped out.

But I think that this is almost certainly system dependent.

**Comment by jchodera** [comment]
True, but if you don't have du/dl because you are not doing TI, or if you have collected du/dl from a replica-exchange simulation, this makes no sense.

**Comment by jchodera** [comment]
> Using u in a system where the not-alchemically-changing energy large can result in spuriously low values, as the time correlation for the alchemical part can be slower than the energy sloshing, and get swamped out.

Do you have any actual data where this is the case?  I'm just curious to find concrete examples.

**Comment by mrshirts** [comment]
You're definitely right about replica exchange -- we need a better way to
decorrelate replica exchange.  I'm not sure the super-potential is correct,
for the same reasons I list above -- but what I'm listing here is
definitely not right either.

For not doing TI, then decorrelating u_{i+1}-u_{i} is the right thing, not
u (with temperature reweighting, u is the right thing).

**Comment by jchodera** [comment]
> For not doing TI, then decorrelating u_{i+1}-u_{i} is the right thing, not u (with temperature reweighting, u is the right thing).

For non-TI, I highly doubt this is any more correct than using the superensemble u, since all differences u_{k}-u_{l} go into MBAR.  Why would u_{i+1}-u{i} be privileged and special in that case?

**Comment by jchodera** [comment]
There is actually a way to test and compare all of these ideas by testing them all on hundreds of replicates of the same simulation, or uncorrelated parts of the same simulation, as in [here](http://dx.doi.org/10.1101/021659).

Are you actively doing these tests, or is this something in need of someone to test?

**Comment by mrshirts** [comment]
>
> For non-TI, I highly doubt this is any more correct than using the
> superensemble u, since all differences u_{k}-u_{l} go into MBAR. Why would
> u_{i+1}-u{i} be privileged and special in that case?
>
I'm restricting now to not replica exchange, since that is much harder.

1) u_{i+1} - u_{i} are generally going to contribute significantly more
than u_{i+j}-u_i are. These energies contribute much more to the weights
than the other energies do (ASSUMING relatively wide spacing, which is
usual, but not always the case. So it's not perfect -- we do need theories
for multiple ensembles).

2)  But even u_{k}-u_{0} excludes the nonalchemical calculations.  So the
full u contains a lot of noise.

In terms of multiple ensemble theories - Perhaps we could use maximum AC
over all j for [u_{i+j}-u_i].  Each of these would be highly correlated
(same configurations).  But I'm pretty sure this wouldn't work that well,
from our own experience.  If you are doing solvation of small molecule,
first turning off charge and then turning off vdW and are near the
decoupled endstate, then it's only interacting weakly with neighbors. The
u_{K-1}-u_K (to slightly stronger vdw) decorrelates quickly. But u{0}-u_K
has very long correlation times, since there you are calculating the full
charge potential, which has gigantic shot noise (going up to 100000's
kj/mol), but those numbers don't contribute at all to the calculation,
since the weights are infintesimal.

**Comment by mrshirts** [comment]
> There is actually a way to test and compare all of these ideas by testing
them all on hundreds of replicates of the same simulation

I am aware that testing hypotheses is something that can be done in science.

The last time we tried direct comparison with 1000's of replicates, it took
a student months (see 2011 benchmark paper with Himanshu). It should be
planned and done at some point, but all my students are committed to other
projects at the current time.
In that case, we intentionally overestimated the correlation time, since
the main test was how well the formulas worked (with uncorrelated data,
very well -- so it comes down to the question of what uncorrelated data is).

**Comment by mrshirts** [comment]
I think there's two issues here -- avoiding doing something definitely wrong + adding something that works at all vs. figuring out best practices.  I think both need to be done, though the time scales are different.

**Comment by davidlmobley** [comment]
Hi, guys. Sorry for the delay - crazy week on this end led to huge e-mail/issue backlog. Let me respond to a couple issues:

>>Using u in a system where the not-alchemically-changing energy large can result in spuriously low values, as the time correlation for the alchemical part can be slower than the energy sloshing, and get swamped out.

>Do you have any actual data where this is the case? I'm just curious to find concrete examples.

We've definitely observed this many times. Unfortunately I don't have a good example on hand right now. It wasn't something I was aware of anyone being particularly interested in at the time, and it seemed intuitive to us (i.e. that dU/dlambda would be a lot less sensitive to distant conformational fluctuations in the system than the total potential energy would be) and something that had already been published on[1]. So we just made note of it and proceeded. As I said though, this is why we've used dU/dlambda for the correlation analysis, though this obviously needs to be replaced when we don't have TI. 

[1] When I say it's already been published on, I don't mean this *exact* issue - but effectively this issue has been addressed peripherally, because it's related to the whole reason why TI and BAR/MBAR work on reasonably short simulations in the first place. If you think about calculating the change in enthalpy for some process, the total energy of the system fluctuates so much that for reasonable-sized (i.e. protein+solvent) systems it's impossible to calculate an enthalpy change with reasonable error bars with simulations of reasonable length. Yet free energy changes via TI or BAR/MBAR/Zwanzig are accessible. Why? Because the large fluctuations in the potential energy due to distant fluctuations in the system don't affect dU/dlambda or the energy differences [u_{i+j}-u_i] nearly as much as they affect the total potential energy. So these fluctuations end up mostly dropping out and we can still get fairly precise free energy differences even though we can't get precise potential energy differences. I believe this ALSO implies that the correlation time for dU/dlambda or [u_{i+j}-u_i] will typically be longer than for the total potential energy, since those fast/distant fluctuations (i.e. water rearrangements far from the binding site) aren't present to lower the correlation time. 

>I think there's two issues here -- avoiding doing something definitely wrong + adding something that works at all vs. figuring out best practices. I think both need to be done, though the time scales are different.

I agree with Michael here. Specifically, I think in the short term what we ought to be doing is:
* decorrelate based on dU/dlambda when that data is available
* decorrelate based on [u_{i+j}-u_i] when it's not

Longer-term, POSSIBLY someone should study the differences between these strategies relative to decorrelating based on the whole potential, in the interests of nailing down what's really best vs what works and makes sense. But as I said, we saw a variety of instances where the correlation time was shorter based on the whole potential/reduced potential than it was based on dU/dlambda, and intuitively this made sense to us, so we pushed ahead.
